### PR TITLE
Fix player drop handling.

### LINKED ISSUE #10614: Some players have connection issues when they leave
Like Scuba Steve, many times I play with him, when he leaves when the game is ended, he got connection issues, then we have to wait for him to be dropped. If anyone exit while that countdown is, then it crash.
We should find out what systems he uses..

I see this also when he is winning, so I don't think its wifi-dc, etc...
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30237206-some-players-have-connection-issues-when-they-leave?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #10615: "Please do not leave, user xxx is beeing dropped", user drops while=freeze game
We should change the info, since new players drop when one player have connection issues, and then the game stalls.

Se allso #10614 
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30237284-please-do-not-leave-user-xxx-is-beeing-dropped-user-drops-while-freeze-game?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
Fixes #10614 and possibly also the desyncs that sank #10787 and the ai-takeover PR.
Closes #10615.

### CODE REVIEW

**Comment by pchote** [comment]
This affects more than just player lag situations (see also #5027), so I suggest this for the changelog:
* Fixed an issue that sometimes stalled multiplayer servers when a player leaves.

The easiest way to test this PR is to run multiple clients and then pause one of them by sending a `SIGSTOP` signal (`^z` in the terminal window, or `kill -SIGSTOP <pid>` from another terminal).  The client can be resumed by typing `fg` in the terminal or `kill -SIGCONT <pid>` from another.

**Review by RoosterDragon — APPROVED** [review]
I can confirm the bug reproduces with the above method and that this PR fixes it.

File: OpenRA.Game/Server/Connection.cs [hunk 1]
```diff
@@ -104,8 +104,10 @@ public void ReadData(Server server)
 
 						case ReceiveState.Data:
 							{
+								if (Frame > MostRecentFrame)
+									MostRecentFrame = Frame;
```
Comment (RoosterDragon) [inline OpenRA.Game/Server/Connection.cs]: I often like to keep the variable being assigned to on the left in the conditional in expressions like this, i.e.
```c#
if (MostRecentFrame < Frame)
    MostRecentFrame = Frame;
```

That's probably just a matter of taste though.

Comment (pchote) [inline OpenRA.Game/Server/Connection.cs]: I agree and will change it.

Comment (pchote) [inline OpenRA.Game/Server/Connection.cs]: Fixed.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/deac41cf91c70a36a9d96a448079ab54257b0843...822b82fde176bb752b53f4ec10a60bce7a6af9d9)

**Pull request merged**